backend/app/core/security.py

```
@@ -671,3 +671,138 @@ def get_current_user_flexible(
+            username_from_key = None
+            if "#" in api_key:
+                parts = api_key.split("#", 1)
+                actual_api_key = parts[0]
+                username_from_key = parts[1] if len(parts) > 1 and parts[1] else None
+
+            key_hash = hashlib.sha256(actual_api_key.encode()).hexdigest()
+            api_key_record = (
+                db.query(APIKey)
+                .filter(
+                    APIKey.key_hash == key_hash,
+                    APIKey.is_active == True,
```

Ruff (0.14.14)

[error] Avoid equality comparisons to `True`; use `APIKey.is_active:` for truth checks

Replace with `APIKey.is_active`

(E712)